pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -244,6 +252,73 @@ type ReconcileClusterSync struct {
 	// remoteClusterAPIClientBuilder is a function pointer to the function that gets a builder for building a client
 	// for the remote cluster's API server
 	remoteClusterAPIClientBuilder func(cd *hivev1.ClusterDeployment) remoteclient.Builder
+
+	podname    string
+	podnameSet bool
+}
+
+// isSyncAssignedToMe determines if this instance of the controller is assigned to the resource being sync'd
+func (r *ReconcileClusterSync) isSyncAssignedToMe(cd *hivev1.ClusterDeployment, logger log.FieldLogger) (bool, error) {
+	hiveNS := controllerutils.GetHiveNamespace()
+
+	logger.Debug("Getting statefulset")
+	sts := &appsv1.StatefulSet{}
+	err := r.Get(context.Background(), types.NamespacedName{Namespace: hiveNS, Name: stsName}, sts)
+	if err != nil {
+		logger.WithError(err).WithField("hiveNS", hiveNS).Error("error getting statefulset.")
+		return false, err
+	}
+
+	logger.Debug("Ensuring replicas is set")
+	if sts.Spec.Replicas == nil {
+		return false, errors.New("sts.Spec.Replicas not set")
+	}
+
+	if sts.Status.CurrentReplicas != *sts.Spec.Replicas {
+		// This ensures that we don't have partial syncing which may make it seem like things are working.
+		// TODO: Remove this once https://issues.redhat.com/browse/CO-1268 is completed as this should no longer be needed.
+		return false, fmt.Errorf("statefulset replica count is off. current: %v  expected: %v", sts.Status.CurrentReplicas, *sts.Spec.Replicas)
+	}
+
+	if !r.podnameSet {
+		return false, errors.New("environment variable HIVE_CLUSTERSYNC_POD_NAME not set")
+	}
+
+	if cd == nil {
+		return false, errors.New("clusterdeployment is nil")
+	}
+
+	logger.Debug("Getting uid for hashing")
+	var uidAsBigInt big.Int
+	hexUID := strings.Replace(string(cd.UID), "-", "", 4)
+	logger.Debugf("hexUID: %+v", hexUID)
+	uidAsBigInt.SetString(hexUID, 16)
```

Comment:
this seems like quite a bit of assumption here, 
- one that there are 4 sections
- and each one is a hex number.

Since these UIDs are all UUIDv4 can we link in the commit/here why these assumptions hold true?